python/ray/train/examples/train_linear_dataset_example.py

```diff
@@ -122,6 +113,13 @@ def train_func(config):
             feature_column_dtypes=[torch.float],
             batch_size=batch_size)
 
+        rank = train.local_rank()
+
+        if torch.cuda.is_available():
+            device = torch.device("cpu")
+        else:
+            device = torch.device(f"cuda:{rank}")
```

Comment:
Example for above comment, looks like this is reversed 😄

Reply:
Also wondering if there's a way to encapsulate this into the utilities, so that the user doesn't need to manually handle the device anywhere.

Reply:
For this example we're using the `device` to move the tensors, but I think the `_WrappedDataLoader` handles this?

Reply:
Here we're not using `prepare_data_loader` or a `DataLoader` at all since it's coming from a Ray Dataset shard.

Reply:
Not sure what the best way to handle Ray Dataset/to_torch usability would be, but left that as out of scope for this PR

Reply:
Gotcha that makes sense!